test/ExecutionTests.jl

```diff
@@ -148,12 +148,38 @@ tune!(b_pass)
 # warmup #
 ###########
 
-p = params(warmup(@benchmarkable sin(1)))
+@test_deprecated warmup(@benchmarkable sin(1))
+
+is_warm = false
+function needs_warm()
+    global is_warm
+    if is_warm
+        sleep(0.1)
+    else
+        sleep(2)
+        is_warm = true
+    end
+end
+
+w = @benchmarkable needs_warm()
+w.params.seconds = 1
+
+#test that all measurements from lineartrial used in tune! are warm
+is_warm = false
+@test maximum(BenchmarkTools.lineartrial(w, w.params)) < 1e9
+
+#test that run warms up the benchmark
+tune!(w)
```

Comment:
why `tune!` here?

Reply:
The only reason I called `tune!` here is that we never previously called tune on the benchmark. We tested that `lineartrial` warms up, but we never called `tune!`, and I'm used to calling `tune!` first. Would you rather I set evals explicitly here?

Reply:
no that's fine